app/main.py

```diff
@@ -1,34 +1,78 @@
+from typing import List
+
+
 class Deck:
-    def __init__(self, row, column, is_alive=True):
-        pass
+    def __init__(self, row: int, column: int, is_alive: bool = True) -> None:
+        self.row = row
+        self.column = column
+        self.is_alive = is_alive
 
 
 class Ship:
-    def __init__(self, start, end, is_drowned=False):
-        # Create decks and save them to a list `self.decks`
-        pass
+    def __init__(
+            self, start: tuple,
+            end: tuple,
+            is_drowned: bool = False
+    ) -> None:
+
+        self.start = start
+        self.end = end
+        self.is_drowned = is_drowned
+        self.deck = []
+
+        self.create_ship()
+
+    def create_ship(self) -> None:
+        for coord_x in range(
+                self.start[0], self.end[0] + 1
+        ):
 
-    def get_deck(self, row, column):
-        # Find the corresponding deck in the list
-        pass
+            for coord_y in range(self.start[1], self.end[1] + 1):
+                if coord_x == self.start[0] or coord_y == self.start[1]:
+                    self.deck.append(Deck(coord_x, coord_y))
 
-    def fire(self, row, column):
-        # Change the `is_alive` status of the deck
-        # And update the `is_drowned` value if it's needed
-        pass
+    def get_deck(self, row: tuple, column: tuple) -> tuple:
+        for deck in self.deck:
+            if deck.row == row and deck.column == column:
+                return deck
+
+    def fire(self, row: tuple, column: tuple) -> None:
+        deck = self.get_deck(row, column)
+        if deck is not None:
+            deck.is_alive = False
+            if all(not d.is_alive for d in self.deck):
+                self.is_drowned = True
 
 
 class Battleship:
-    def __init__(self, ships):
-        # Create a dict `self.field`.
-        # Its keys are tuples - the coordinates of the non-empty cells,
-        # A value for each cell is a reference to the ship
-        # which is located in it
-        pass
-
-    def fire(self, location: tuple):
-        # This function should check whether the location
-        # is a key in the `self.field`
-        # If it is, then it should check if this cell is the last alive
-        # in the ship or not.
-        pass
+    def __init__(self, ships: List[tuple]) -> None:
+        self.ships = ships
+        self.field = {}
+        self.game_field = self.create_battle_field(ships)
+        self.placement_of_ships()
+
+    def fire(self, location: tuple) -> str:
+        if location not in self.field:
+            return "Miss!"
+        if location in self.field:
+            ship = self.field[location]
+            ship.fire(*location)
+            return "Sunk!" if ship.is_drowned else "Hit!"
+
+    def create_battle_field(self, ships: List[tuple]) -> List[list[str]]:
+        return [["~" for _ in range(10)] for _ in range(10)]
```

Comment:
if you want to implement printing logic, you have to implement it for both wounded and killed ships, and now it only works for starting the game

Reply:
leave the create_battle_field method in which implement the self.field filling logic,
and also create a print_field method in which you implement all the printing logic